src/Motor.Extensions.Diagnostics.Metrics/AutoObserveStopwatch.cs

```diff
@@ -0,0 +1,36 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Prometheus.Client;
+
+namespace Motor.Extensions.Diagnostics.Metrics
+{
+    public class AutoObserveStopwatch : IDisposable
+    {
+        private readonly Stopwatch _watch;
+        private readonly ISummary? _unlabeledSummary;
```

Comment:
I would prefer to use `IValueObserver` instead of `ISummary` because this also covers histograms but only summaries.